andela/reaction-commerce
# #133002633 user should be able to make payments with wallet and paystack

#### What does this PR do?
Implements features for Users to make payments with Wallet and Paystack payment methods.

#### Description of Task to be completed?
- Integrate Paystack as a payment method
- Create Wallet for all users
- Refund Wallet on cancel order
- Fund Wallet via Paystack
-  Integrate Wallet as a Payment Method
- Resolve Currency conflicts

#### How should this be manually tested?
- On the Admin dashboard, Wallet and Paystack can be enabled or disabled
- On the checkout page, if enabled, Wallet and Payments are present as payment options

>*Paystack*
After filling in the required email and phone number fields, click the button ```Pay with Paystack```. A pop up modal appears where  card details get filled in. Click the button ```Pay```.

> *Wallet*
If Wallet is not empty click button ```Pay``` to pay with Wallet.
Click button ```Top up Wallet``` to display a short form. Fill in required fields to and click ```Fund Wallet``` to top up Wallet with Paystack.
A pop up modal appears where card details gets field in. 

#### What are the relevant pivotal tracker stories?
133002633

#### Screenshots (if appropriate)
![screenshot from 2016-11-19 23-41-55](https://cloud.githubusercontent.com/assets/17247842/20458957/c519bd88-aeb2-11e6-928f-3cd871f3e714.png)
![screenshot from 2016-11-19 23-44-50](https://cloud.githubusercontent.com/assets/17247842/20458959/cc400f54-aeb2-11e6-80d6-e47c78a54302.png)
![screenshot from 2016-11-19 23-45-18](https://cloud.githubusercontent.com/assets/17247842/20458970/edc32ff8-aeb2-11e6-825a-f67290e6fa09.png)
![screenshot from 2016-11-19 23-45-24](https://cloud.githubusercontent.com/assets/17247842/20458971/f3f04618-aeb2-11e6-8ada-2bb8e10685d7.png)
![screenshot from 2016-11-19 23-46-16](https://cloud.githubusercontent.com/assets/17247842/20458974/0019a254-aeb3-11e6-8a52-f0b63807ad0a.png)
![screenshot from 2016-11-19 23-46-54](https://cloud.githubusercontent.com/assets/17247842/20458988/4ed8214a-aeb3-11e6-8124-76a3e377c3f6.png)
